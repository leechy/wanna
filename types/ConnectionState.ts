export type ConnectionState = {
  isConnected: boolean;
  lastConnected: string | null;
  errors: string[];
};
