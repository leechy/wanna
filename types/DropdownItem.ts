import { SvgProps } from 'react-native-svg';

export type DropdownItem = {
  label: string;
  onPress?: (item?: DropdownItem) => void;
  icon?: React.FC<SvgProps>;
  color?: string;
};
