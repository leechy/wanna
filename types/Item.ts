export type ItemUser = {
  uid: string;
  names: string;
};

export type Item = {
  itemId: string;
  listItemId: string;
  listId: string;

  name: string;
  type: 'item' | 'task';
  units?: string;
  quantity: string | number;
  deadline?: string;

  active?: boolean;
  public?: boolean;

  ongoing?: boolean;
  assignee?: string | null;
  assigneeId?: string | null;
  completed?: boolean;
  completedAt?: string | null;

  sortOrder?: number;

  deleted?: boolean;
  createdAt?: string;
  updatedAt?: string;
};
