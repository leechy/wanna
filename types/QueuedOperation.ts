export interface QueuedOperation {
  id: string;
  timestamp: number;
  event: string;
  data: any;
}
