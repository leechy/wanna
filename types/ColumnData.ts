import { SvgProps } from 'react-native-svg';
import { ListItem } from '@/types/listItem';

export type ColumnData = {
  title: string;
  color?: string;
  action?: React.ReactNode;
  newItemLabel?: string;
  newItemHandler?: () => void;
  newItemIcon?: React.FC<SvgProps>;
  actionHandler?: (item: ListItem) => void;
  checkboxHandler?: (item: ListItem) => void;
  longPressHandler?: (item: ListItem) => void;
  editHandler?: (item: ListItem) => void;
  deleteHandler?: (item: ListItem) => void;
  resetHandler?: (item: ListItem) => void;
  items: ListItem[];
  onToggle?: () => void;
  isOpen?: boolean;
  showEmpty?: boolean;
  emptyText?: string;
  onNew?: () => void;
};
