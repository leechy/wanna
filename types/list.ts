import { Item } from '@/types/Item';

export type List = {
  listId: string;
  shareId: string;
  name: string;
  type: 'project' | 'shopping-list' | 'recipe';
  deadline?: string | null;

  active: boolean;
  completed: boolean;
  completedAt?: string;
  hideCompleted?: boolean;

  sortOrder?: number;
  notifyOnListShared?: boolean;
  notifyOnListItemsUpdate?: boolean;
  notifyOnItemStateUpdate?: boolean;

  deleted: boolean;
  createdAt: string;
  updatedAt: string;

  users?: ListUser[];
  listItems?: Item[];
};

export type ListUser = {
  uid: string;
  names: string;
};

export type ListState = {
  [id: string]: List;
};
