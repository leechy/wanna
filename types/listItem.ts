import { Item } from '@/types/Item';

export type ListItem = {
  type: // item types
  | 'new'
    | 'item'
    | 'task'
    // list types
    | 'shopping-list'
    | 'project'
    | 'recipe'
    // other types
    | 'contact'
    | 'group';
  id: string;
  label: string;
  list?: string;
  shared?: string[];
  deadline?: string | number | null;
  quantity?: number;
  ongoing?: boolean | number; // replacement for the progress property
  completed?: boolean | number;
  updatedAt?: string;
  sortOrder?: number;
  completedAt?: string | null;
  assignee?: string | null;
  assigneeId?: string | null;
};

export function convertItemsToListItems(items?: Item[] | null): ListItem[] {
  if (!items) {
    return [];
  }

  return items.map((item) => ({
    type: 'item',
    id: item.listItemId,
    label: item.name,
    listId: item.listId,
    deadline: item.deadline,
    quantity: item.quantity,
    ongoing: item.ongoing,
    assignee: item.assignee,
    assigneeId: item.assigneeId,
    completed: item.completed,
    completedAt: item.completedAt,
    updatedAt: item.updatedAt,
    sortOrder: item.sortOrder,
  }));
}
