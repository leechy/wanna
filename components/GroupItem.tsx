// hooks
import { useThemeColor } from '@/hooks/useThemeColor';

// components
import { StyleProp, StyleSheet, Text, TouchableOpacity, View, ViewStyle } from 'react-native';

// styles and icons
import { globalStyles } from '@/constants/GlobalStyles';
import PurchaseIcon from '@/assets/symbols/bag-fill.svg';

// types
import { ListItem } from '@/types/listItem';

interface GroupItemProps {
  item: ListItem;
  actionHandler?: (item: ListItem) => void;
  itemBorderRadius: StyleProp<ViewStyle>;
}

export default function GroupItem({ item, itemBorderRadius, actionHandler }: GroupItemProps) {
  const backgroundColor = useThemeColor({}, 'listBackground');
  const inactiveColor = useThemeColor({}, 'inactive');

  function onItemAction() {
    console.log('ItemsList Action item', item, actionHandler);
    actionHandler?.(item);
  }

  return (
    <View style={[styles.groupItem, { backgroundColor, minHeight: 20 }, itemBorderRadius]}>
      <View style={globalStyles.listItemLeadButton}>
        <PurchaseIcon width={20} height={20} color={inactiveColor} />
      </View>
      <View style={globalStyles.itemListLabelContainer}>
        <Text
          style={[globalStyles.itemListlabel, { color: inactiveColor, fontSize: 14 }]}
          adjustsFontSizeToFit={true}
          numberOfLines={2}
        >
          {item.label}
        </Text>
      </View>
    </View>
  );
}

const styles = StyleSheet.create({
  groupItem: {
    ...globalStyles.listItem,
    flexDirection: 'row',
    justifyContent: 'space-between',
  },
});
