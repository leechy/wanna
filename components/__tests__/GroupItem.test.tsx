import React from 'react';
import { render } from '@testing-library/react-native';
import GroupItem from '../GroupItem';
import { ListItem } from '@/types/listItem';

jest.mock('../../assets/symbols/bag-fill.svg', () => 'PurchaseIcon');

describe('GroupItem', () => {
  it('renders the item correctly', () => {
    const item: ListItem = { id: '1', type: 'group', label: 'Groceries' };
    const { getByText } = render(<GroupItem item={item} itemBorderRadius={{}} />);
    expect(getByText('Groceries')).toBeTruthy();
  });

  it('matches snapshot', () => {
    const item: ListItem = { id: '1', type: 'group', label: 'Groceries' };
    const { toJSON } = render(<GroupItem item={item} itemBorderRadius={{}} />);
    expect(toJSON()).toMatchSnapshot();
  });
});
