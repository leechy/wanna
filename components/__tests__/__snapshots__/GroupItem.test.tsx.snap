// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GroupItem matches snapshot 1`] = `
<View
  style={
    [
      {
        "alignItems": "center",
        "borderBottomWidth": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "minHeight": 48,
        "overflow": "hidden",
      },
      {
        "backgroundColor": "#ffffff",
        "minHeight": 20,
      },
      {},
    ]
  }
>
  <View
    style={
      {
        "padding": 5,
      }
    }
  >
    <PurchaseIcon
      color="#999999"
      height={20}
      width={20}
    />
  </View>
  <View
    style={
      {
        "alignItems": "flex-start",
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "space-between",
        "paddingVertical": 8,
      }
    }
  >
    <Text
      adjustsFontSizeToFit={true}
      numberOfLines={2}
      style={
        [
          {
            "fontFamily": "Montserrat",
            "fontSize": 15,
            "lineHeight": 18,
          },
          {
            "color": "#999999",
            "fontSize": 14,
          },
        ]
      }
    >
      Groceries
    </Text>
  </View>
</View>
`;
